File: go-version/leetcode/LC1656/ordered_stream.go
package LC1656

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/2/24 下午11:08
 * @FilePath: ordered_stream
 * @Description: 设计有序流
 */

type OrderedStream struct {
	ptr    int
	stream []string
}

func Constructor(n int) OrderedStream {
	return OrderedStream{
		stream: make([]string, n+1),
		ptr:    1,
	}
}

func (this *OrderedStream) Insert(idKey int, value string) []string {
	this.stream[idKey] = value
	list := make([]string, 0)
	for this.ptr < len(this.stream) && this.stream[this.ptr] != "" {
		list = append(list, this.stream[this.ptr])
		this.ptr++
	}
	return list
}

File: java-version/src/main/java/com/github/leetcode/LC1656/OrderedStream.java
package com.github.leetcode.LC1656;

import java.util.ArrayList;
import java.util.List;

public class OrderedStream {
    private String[] stream;
    private int ptr;

    public OrderedStream(int n) {
        this.stream = new String[n+1];
        this.ptr = 1;
    }

    public List<String> insert(int idKey, String value) {
        this.stream[idKey] = value;
        List<String> list = new ArrayList<>();
        while (this.ptr < this.stream.length && this.stream[this.ptr] != null) {
            list.add(this.stream[this.ptr++]);
        }
        return list;
    }
}
